### PR TITLE
fix: avoid parsing old value if not exists

### PR DESCRIPTION
**Pull request by prakashchokalingam**
Please refer: https://github.com/PlasmoHQ/storage/issues/6

### CODE REVIEW

**Comment by ColdSauce** [comment]
Hey, @prakashchokalingam thanks for the pull request, and for helping us fix this bug! 

I think this fix can be condensed to one line:

```
oldValue: JSON.parse(changes[key].oldValue || null)
```

`JSON.parse` doesn't know what undefined means but does know what null means as it's a valid JSON identifier.

**Comment by prakashchokalingam** [comment]
Hi,  @ColdSauce  Thought the same. Yet `JSON.parse(null)` will add the parse computation, Hence skipping the JSON parse itself.

I am fine with both approaches.

**Comment by louisgv** [comment]
Hmm I do think that keeping it undefined (i.e, when it is not initialized) would be cleaner. `undefined` is a runtime concept vs `null` is an user concept. If something is not yet set, it should stay `undefined`

File: src/index.ts [hunk 1]
```diff
@@ -169,13 +169,26 @@ export class Storage {
       for (const key of relevantKeyList) {
         callbackMap[key](
           {
-            newValue: JSON.parse(changes[key].newValue),
-            oldValue: JSON.parse(changes[key].oldValue)
+            newValue: this.#parseValue(changes[key].newValue),
```
Comment (ColdSauce) [inline src/index.ts:172]: Can you update other JSON.parse values as well @louisgv

**Review by ColdSauce — CHANGES_REQUESTED** [review]

**Review by louisgv — APPROVED** [review]

File: src/index.ts (hunk 1 above)
Comment (louisgv) [reply src/index.ts:172]: wait where?

Comment (ColdSauce) [reply src/index.ts:172]: Like in the other functions in the file that parse the storage val

Comment (louisgv) [reply src/index.ts:172]: let's apply this for the whole file after we merge this imo.

**Review by ColdSauce — APPROVED** [review]

**Comment by prakashchokalingam** [comment]
Thanks @ColdSauce @louisgv

**Pull request merged**